nop/nop_test.go

```diff
@@ -0,0 +1,89 @@
+package nop_test
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/philippgille/gokv"
+	"github.com/philippgille/gokv/nop"
+)
+
+func TestNop(t *testing.T) {
+	t.Parallel()
+
+	var s gokv.Store = nop.NewStore()
+
+	if err := s.Set("foo", 1); err != nil {
+		t.Error(err)
+	}
+
+	var v int
+	found, err := s.Get("foo", &v)
+	if err != nil {
+		t.Error(err)
+	}
+	if found {
+		t.Error("A value was found, but no value was expected")
+	}
+
+	if err := s.Delete("foo"); err != nil {
+		t.Error(err)
+	}
+
+	if err := s.Close(); err != nil {
+		t.Error(err)
+	}
+}
+
+func TestInputValidation(t *testing.T) {
+	t.Parallel()
+
+	var s gokv.Store = nop.NewStore()
+
+	{
+		err := s.Set("", 1)
+		assertEqualError(t, err, errInvalidKey.Error())
+	}
+
+	{
+		err := s.Set("foo", nil)
+		assertEqualError(t, err, errInvalidValue.Error())
+	}
+
+	{
+		var v int
+		found, err := s.Get("", &v)
+		assertEqualError(t, err, errInvalidKey.Error())
+		if found {
+			t.Error("A value was found, but no value was expected")
+		}
+	}
+
+	{
+		found, err := s.Get("foo", nil)
+		assertEqualError(t, err, errInvalidValue.Error())
+		if found {
+			t.Error("A value was found, but no value was expected")
+		}
+	}
+
+	{
+		err := s.Delete("")
+		assertEqualError(t, err, errInvalidKey.Error())
+	}
+}
+
+func assertEqualError(t *testing.T, err error, expectedErrMsg string) {
+	t.Helper()
+
+	if err == nil {
+		t.Error("expect error, got nil")
```

Comment:
Not a blocker, but checking all other `*_test.go` files in this repo, they use either "Expected an error" or "An error was expected". The exact phrasing doesn't matter, but I'd suggest uppercase and past tense for consistency, e.g.

```suggestion
		t.Error("Expected error, got nil")
```

Errors are usually lowercase as they can be chained in a log message (e.g. "couldn't x: failed to y: ...", but here it's a log message and not an error that might be chained, where I like the differentiation.

Again, totally not a blocker though.